File: 0925-long-pressed-name/0925-long-pressed-name.java
class Solution {
    public boolean isLongPressedName(String name, String typed) {
        int l=0;
        int r=0;
        while(r<typed.length() && l<name.length())
        {
            if(name.charAt(l)==typed.charAt(r))
            {
                l++;
                r++;
            }
            else if(r>0 && typed.charAt(r)==typed.charAt(r-1))
            {
                r++;
            }
            else
            {
                return false;
            }

        }
        if(l<name.length())
        {
            return false;
        }
        if(name.length()>typed.length())
        {
            return false;
        }
        while(r<typed.length())
        {
            if(typed.charAt(r)!=typed.charAt(r-1))
            {
                return false;
            }
            r++;
        }
        return true;
        
    }
}

File: 0925-long-pressed-name/0925-long-pressed-name.py
class Solution:
    def isLongPressedName(self, name: str, typed: str) -> bool:
        ptr1=0
        ptr2=0
        if(len(typed)<=len(name)):
            if(name!=typed):
                return False       
        while(ptr1<len(name)and ptr2<len(typed)):
            if(name[ptr1]==typed[ptr2]):
                ptr1+=1
                ptr2+=1
            else:
                if(ptr2>0 and typed[ptr2]==typed[ptr2-1]):
                    ptr2+=1
                else:
                    return False
        if(ptr1<len(name)):
            return False
        while(ptr2<len(typed)):
            if(name[-1]!=typed[ptr2]):
                return False
            ptr2+=1
        return True
        